pkg/meta/meta.go

```diff
@@ -63,6 +65,18 @@ func ReferenceTo(o metav1.Object, of schema.GroupVersionKind) *corev1.ObjectRefe
 	}
 }
 
+// TypedReferenceTo returns a typed object reference to the supplied object,
+// presumed to be of the supplied group, version, and kind.
+func TypedReferenceTo(o metav1.Object, of schema.GroupVersionKind) *v1alpha1.TypedReference {
+	v, k := of.ToAPIVersionAndKind()
+	return &v1alpha1.TypedReference{
+		APIVersion: v,
+		Kind:       k,
+		Name:       o.GetName(),
+		UID:        o.GetUID(),
```

Comment:
What do you think about omitting the UID when this helper function is used? Including it would better match the existing `ReferenceTo` implementation, but I think in most cases we don't want or need to include the UID in references. I believe they're typically purely informational (nothing actually verifies that the referenced object has a matching UID), and I suspect we don't have logic that updates the reference UID when an object is recreated (e.g. during a backup and restore).

Reply:
So in the case I am using this, we are actually updating the status with a reference to the most recently gotten object, so while the UID could be stale for a while, it will eventually get updated. I would prefer to keep it if all the same to you.

Reply:
I'm not super convinced there's value in including the UID, but given we more often do so today I'm happy to leave that for a future change.

Reply:
I think that's fair, but I think it should be removed from `TypedReference` as opposed to just not returning it here.

Reply:
Would you like for me to go ahead and remove it from both?

Reply:
As you mentioned out of band, this function is now being used for owner references, which require UIDs. I'm happy to leave the UIDs in. I suspect doing so will continue a pattern of including UIDs in some contexts where they're not particularly useful, but that's not a regression compared to the existing `ReferenceTo`.